TF_Settings_Web/src/Components/Pages/Camera/Masking/MaskingScreen.tsx

```diff
@@ -135,60 +134,79 @@ const MaskingScreen = () => {
+    //     if (!canvasContextRef.current) return;
+    //     if (isFrameProcessing.current || !allowImages.current) return;
+
+    //     const data = event.data as ArrayBuffer;
+    //     if (!data) return;
+
+    //     const dataIdentifier = new Uint8Array(data, 0, 1)[0];
+    //     if (dataIdentifier === 1) {
+    //         isFrameProcessing.current = true;
+    //         successfullySubscribed.current = true;
+
+    //         updateCanvas(
+    //             data,
+    //             canvasContextRef.current,
+    //             mainLens.current,
+    //             isCamReversed.current,
+    //             showOverexposed.current
+    //         );
+    //         frameTimeoutRef.current = window.setTimeout(() => {
+    //             isFrameProcessing.current = false;
+    //         }, FRAME_PROCESSING_TIMEOUT);
+    //     } else if (!successfullySubscribed.current) {
+    //         socket.send(JSON.stringify({ type: 'SubscribeImageStreaming' }));
+    //     }
+    // };
+
+    const handleTFInput = (evt: CustomEvent<ArrayBuffer>): void => {
+        if (isHandProcessing.current) return;
 
-    const handleMessage = (socket: WebSocket, event: MessageEvent) => {
-        if (!canvasContextRef.current) return;
-        if (isFrameProcessing.current || !allowImages.current) return;
-
-        const data = event.data as ArrayBuffer;
-        if (!data) return;
-
-        const dataIdentifier = new Uint8Array(data, 0, 1)[0];
-        if (dataIdentifier === 1) {
-            isFrameProcessing.current = true;
-            successfullySubscribed.current = true;
-
-            updateCanvas(
-                data,
-                canvasContextRef.current,
-                mainLens.current,
-                isCamReversed.current,
-                showOverexposed.current
-            );
-            frameTimeoutRef.current = window.setTimeout(() => {
-                isFrameProcessing.current = false;
-            }, FRAME_PROCESSING_TIMEOUT);
-        } else if (!successfullySubscribed.current) {
-            socket.send(JSON.stringify({ type: 'SubscribeImageStreaming' }));
-        }
-    };
+        isHandProcessing.current = true;
 
-    const handleTFInput = (evt: CustomEvent<HandFrame>): void => {
-        if (isHandProcessing.current || !successfullySubscribed.current) return;
+        try {
+            const buffer = evt.detail;
+            const imageArraySize = new Int32Array(buffer, 4, 8)[0];
+
+            if (imageArraySize > 0 && canvasContextRef.current) {
+                updateCanvas(
+                    buffer.slice(8, imageArraySize),
+                    canvasContextRef.current,
+                    isCamReversed.current,
+                    showOverexposed.current
+                );
+            }
 
-        isHandProcessing.current = true;
+            const handsJson = String.fromCharCode.apply(null, [...new Uint8Array(buffer, 8 + imageArraySize)]);
```

Comment:
Don't need to spread back into an array and use apply. Can just spread into fromCharCode()

Reply:
Updated this